students/rod_musser/lesson07/assignment/parallel.py

```diff
@@ -0,0 +1,230 @@
+                              QTY_AVAIL: prod[QTY_AVAIL]}
+            available_products[prod[PRODUCT_ID]] = prod_info_dict
+    logger.info(available_products)
+    return available_products
+
+
+def show_rentals(product_id):
+    """
+    Returns a dict of users that have rented this product id
+    :param product_id: The id of the product that you want a list of rental users
+    """
+    logger = utilities.configure_logger('default', 'mongo.log')
+    mongo = MongoDBConnection()
+    logger.debug('Connected to Mongo DB')
+    cust_rentals = {}
+
+    with mongo:
+        # mongodb database; it all starts here
+        database = mongo.connection.media
+        logger.debug('Got the database.')
+        result = database[CUSTOMER].aggregate([
+            {'$lookup': {
+                'from': RENTAL,
+                'let': {USER_ID: "$user_id"},
+                'pipeline': [
+                    {'$match': {
+                        '$expr': {
+                            '$and':
+                                [{'$eq': ["$" + PRODUCT_ID, product_id]},
+                                 {'$eq': ["$" + USER_ID, '$$user_id']}]
+                        }
+                    }
+                    }
+                ],
+                'as': "cust_rentals"
+            }
+            }, {'$unwind': {
+                'path': "$cust_rentals",
+                'preserveNullAndEmptyArrays': False
+            }}
+        ])
+        for cust in result:
+            logger.debug(cust)
+            cust_info_dict = {NAME: cust[NAME],
+                              ADDRESS: cust[ADDRESS],
+                              PHONE_NUMBER: cust[PHONE_NUMBER],
+                              EMAIL: cust[EMAIL]}
+            cust_rentals[cust[USER_ID]] = cust_info_dict
+
+    logger.info(cust_rentals)
+    print('Finished Loading Data')
+    return cust_rentals
+
+
+def drop_collections():
+    """
+    Clean up the database
+    """
+    logger = utilities.configure_logger('default', 'mongo.log')
+    mongo = MongoDBConnection()
+    logger.debug('Connected to Mongo DB')
+
+    with mongo:
+        database = mongo.connection.media
+        logger.debug('Got the database.')
+
+        customer = database[CUSTOMER]
+        customer.drop()
+
+        product = database[PRODUCT]
+        product.drop()
+
+        rental = database[RENTAL]
+        rental.drop()
+
+
+def main():
+    """
+    Loads product, customer, and rental data
+    """
+    drop_collections()
+    threads = []
+    data = ['products.csv', 'customers.csv', 'rentals.csv']
+    tables = [PRODUCT, CUSTOMER, RENTAL]
+    fields = [PRODUCT_FIELDS, CUSTOMER_FIELDS, RENTAL_FIELDS]
+    results = queue.Queue()
+
+    def worker(*args):
+        results.put(import_data(*args))
+
+    for i in range(3):
+        thread = threading.Thread(target=worker, args=('./data', data[i], tables[i], fields[i]))
+        thread.start()
+        threads.append(thread)
```

Comment:
You are launching the threads, but you are not using any mechanism to ensure your threads have finished before you continue. Check the course materials on the _join()_ method.